src/core/document.js

```diff
@@ -308,29 +307,112 @@ var Page = (function PageClosure() {
+      });
     },
 
     get annotations() {
-      var annotations = [];
+      var AnnotationWorkerTask = (function AnnotationWorkerTaskClosure() {
+        function AnnotationWorkerTask(name) {
+          this.name = name;
+          this.terminated = false;
+          this.capability = createPromiseCapability();
+        }
+
+        AnnotationWorkerTask.prototype = {
+          get finished() {
+            return this.capability.promise;
+          },
+
+          finish() {
+            this.capability.resolve();
+          },
+
+          terminate() {
+            this.terminated = true;
+          },
+
+          ensureNotTerminated() {
+            if (this.terminated) {
+              throw new Error('Annotation worker task was terminated');
+            }
+          },
+        };
+
+        return AnnotationWorkerTask;
+      })();
+
+      // create a blank annotation fonts array
+      // if it's not initialized yet
+      if (this.pdfManager && this.pdfManager.pdfDocument &&
+          this.pdfManager.pdfDocument.acroForm &&
+          !this.pdfManager.pdfDocument.acroForm.annotationFonts) {
+        this.pdfManager.pdfDocument.acroForm.annotationFonts = [];
+      }
+
+      var task = new AnnotationWorkerTask(
+        'GetAnnotationAppereances: page ' + this.pageIndex);
+
+      var handler = {};
+
+      var self = this;
+
+      handler.send = function (actionname, data) {
+        if (self.pdfManager && self.pdfManager.pdfDocument &&
+            self.pdfManager.pdfDocument.acroForm) {
+          self.pdfManager.pdfDocument.acroForm.annotationFonts.push(data);
+        }
+      };
+
+      var pageNum = this.pageIndex + 1;
+      var start = Date.now();
+
+      var partialEvaluator = new PartialEvaluator({
+        pdfManager: this.pdfManager,
+        xref: this.xref,
+        handler,
+        pageIndex: this.pageIndex,
+        idFactory: this.idFactory,
+        fontCache: this.fontCache,
+        builtInCMapCache: this.builtInCMapCache,
+        options: this.evaluatorOptions,
+      });
+
       var annotationRefs = this.getInheritedPageProp('Annots') || [];
+      var annotationsPromise = [];
       for (var i = 0, n = annotationRefs.length; i < n; ++i) {
         var annotationRef = annotationRefs[i];
-        var annotation = AnnotationFactory.create(this.xref, annotationRef,
-                                                  this.pdfManager,
-                                                  this.idFactory);
-        if (annotation) {
-          annotations.push(annotation);
-        }
+          annotationsPromise.push(AnnotationFactory.create(
+          this.xref,
+          annotationRef,
+          this.pdfManager,
+          this.idFactory,
+          partialEvaluator,
+          task
+        ));
       }
-      return shadow(this, 'annotations', annotations);
+
+      return shadow(this, 'annotations',
+        Promise.all(annotationsPromise)).then(function (annotations) {
```

Comment:
I don't think we need to do all the logging for this worker task. Let's keep things simple here and remove it.

Reply:
Ok, done, thank you